src/target/riscv/riscv-013_reg.c

```diff
@@ -92,26 +93,165 @@ static const struct reg_arch_type *riscv013_gdb_regno_reg_type(uint32_t regno)
 	return &riscv011_reg_type;
 }
 
-static int riscv013_init_reg(struct target *target, uint32_t regno)
+static int init_cache_entry(struct target *target, uint32_t regno)
 {
-	return riscv_reg_impl_init_one(target, regno, riscv013_gdb_regno_reg_type(regno));
+	return riscv_reg_impl_init_cache_entry(target, regno, riscv013_gdb_regno_reg_type(regno));
 }
 
-int riscv013_reg_init_all(struct target *target)
+static int examine_xlen(struct target *target)
 {
-	if (riscv_reg_impl_init_cache(target) != ERROR_OK)
+	RISCV_INFO(r);
+	unsigned int cmderr;
+	const uint32_t command = riscv013_access_register_command(target,
+			GDB_REGNO_S0, /* size */ 64, AC_ACCESS_REGISTER_TRANSFER);
+	int res = riscv013_execute_abstract_command(target, command, &cmderr);
+	if (res == ERROR_OK) {
+		r->xlen = 64;
+		return ERROR_OK;
+	}
+	if (res == ERROR_TIMEOUT_REACHED)
 		return ERROR_FAIL;
+	r->xlen = 32;
+	return ERROR_OK;
+}
+
+/* As defined by RISC-V V extension specification:
+ * https://github.com/riscv/riscv-v-spec/blob/2f68ef7256d6ec53e4d2bd7cb12862f406d64e34/v-spec.adoc?plain=1#L67-L72 */
+#define VLEN_MAX 65536
+
+static int examine_vlenb(struct target *target)
+{
+	RISCV_INFO(r);
+	/* Reading "vlenb" requires "mstatus.vs" to be set, so "mstatus" should
```

Comment:
```suggestion
	RISCV_INFO(r);
	
	/* Reading "vlenb" requires "mstatus.vs" to be set, so "mstatus" should
```
Detail: I believe it would help readability.

Reply:
Addressed.